storage_service/locations/api/sword/helpers.py

```diff
@@ -145,8 +145,7 @@ def spawn_download_task(deposit_uuid, objects, subdir=None):
     """
     Spawn an asynchrnous batch download
     """
-    p = Process(target=_fetch_content, args=(deposit_uuid, objects, subdir))
-    p.start()
+    AsyncManager.run_task(_fetch_content, (deposit_uuid, objects, subdir))
```

Comment:
@sevein are you sure you should be passing a tuple as the second argument to `run_task` here? Doing it the way it's done here, I think it will end up passing a tuple which contains a tuple which dontains `deposit_uuid`, etc. to the `args` kwarg of `threading.Thread`. The signature of `run_task` is `def run_task(task_fn, *args, **kwargs)`.

    >>> def x(a, *args, **kwargs):
    ...   print(a)
    ...   print(args)
    ...   print(kwargs)
    >>> x(1, (2, 3))
    1
    ((2, 3),)
    {}

Reply:
You're right, thanks. It's fixed now.